tests/forms_tests/tests/test_formsets.py

```diff
@@ -1180,51 +1107,53 @@ def test_html_safe(self):
         self.assertEqual(str(formset), formset.__html__())
 
 
-data = {
-    'choices-TOTAL_FORMS': '1',  # the number of forms rendered
-    'choices-INITIAL_FORMS': '0',  # the number of forms with initial data
-    'choices-MIN_NUM_FORMS': '0',  # min number of forms
-    'choices-MAX_NUM_FORMS': '0',  # max number of forms
-    'choices-0-choice': 'Calexico',
-    'choices-0-votes': '100',
-}
-
+class FormsetAsTagTests(SimpleTestCase):
+    def setUp(self):
+        self.data = {
+            'choices-TOTAL_FORMS': '1',
+            'choices-INITIAL_FORMS': '0',
+            'choices-MIN_NUM_FORMS': '0',
+            'choices-MAX_NUM_FORMS': '0',
+            'choices-0-choice': 'Calexico',
+            'choices-0-votes': '100',
+        }
+        self.management_form_html = (
+            '<input type="hidden" name="choices-TOTAL_FORMS" value="1">'
+            '<input type="hidden" name="choices-INITIAL_FORMS" value="0">'
+            '<input type="hidden" name="choices-MIN_NUM_FORMS" value="0">'
+            '<input type="hidden" name="choices-MAX_NUM_FORMS" value="0">'
+        )
 
-class FormsetAsFooTests(SimpleTestCase):
     def test_as_table(self):
-        formset = ChoiceFormSet(data, auto_id=False, prefix='choices')
+        formset = ChoiceFormSet(self.data, auto_id=False, prefix='choices')
```

Comment:
I thought `formset = ChoiceFormSet(self.data, auto_id=False, prefix='choices')` could go in `setUp()` also.